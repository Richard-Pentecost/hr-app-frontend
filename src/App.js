import React, { useState } from 'react';
import { Route, Switch, Redirect, BrowserRouter } from 'react-router-dom';
import Login from './pages/Login';
import Home from './pages/Home';
import EditInformation from './pages/EditInformation';
import ViewEmployee from './pages/ViewEmployee';
import EmployeesList from './pages/EmployeesList';
import CreateEmployee from './pages/CreateEmployee';
import Navbar from './components/Navbar';
import AuthRoute from './components/AuthRoute';
import { ThemeProvider } from '@material-ui/core/styles';
import theme from "./resources/theme";
import './style/App.scss';

const App = () => {
	const [isLoggedIn, setIsLoggedIn] = useState(false);
	const [id, setId] = useState('');
	const [user, setUser] = useState({firstName:'', surname:'', role:'', email:'', telephone:'', doB:'', address:'', nexOfKin:'', permissionsLevel:'', location:''})
	const [currentEmployeeId, setCurrentEmployeeId] = useState('');
	return (
		<ThemeProvider theme={theme}>
			<BrowserRouter>
				<div className="app">
					<Navbar setIsLoggedIn={setIsLoggedIn} isLoggedIn={isLoggedIn} />
					<Switch>
						<Route
							exact 
							path='/' 
							render={props => (isLoggedIn ?
								<Redirect to='/home' /> : 
								<Login {...props} setIsLoggedIn={setIsLoggedIn} setId={setId} />
							)}
						/>

						<AuthRoute 
							exact
							path="/home"
							component={Home}
							isLoggedIn={isLoggedIn}
							id={id}
							user={user} 
							setUser={setUser}
						/>

						<AuthRoute 
							exact
							path='/edit-information'
							component={EditInformation}
							isLoggedIn={isLoggedIn}
							id={id}
							user={user} 
							setUser={setUser}
						/>

						<AuthRoute 
							exact
							path='/employees-list'
							component={EmployeesList}
							isLoggedIn={isLoggedIn}
							setCurrentEmployeeId = {setCurrentEmployeeId}
							currentEmployeeId = {currentEmployeeId}
						/>

						<AuthRoute 
							exact
							path='/create-employee'
							component={CreateEmployee}
							isLoggedIn={isLoggedIn}
							id={id}
							user={user} 
							setUser={setUser}
						/>

						<AuthRoute 
							exact
							path='/view-employee'
							component={ViewEmployee}
							isLoggedIn={isLoggedIn}
							currentEmployeeId = {currentEmployeeId}
						/>
					</Switch>
				</div>
			</BrowserRouter>
		</ThemeProvider>
	);
}

export default App;
