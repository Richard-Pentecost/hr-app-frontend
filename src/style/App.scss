@import './variables.scss';

*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

html {
  font-size: 62.5%; /* defines 1rem as 10px */
  box-sizing: border-box;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  font-weight: 300;
  line-height: 1.5;
  // background-color: $color-blue-3;
  // color: $color-blue-1;
}