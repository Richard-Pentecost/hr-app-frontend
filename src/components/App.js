import React, { useState } from 'react';
import { Route, Switch, Redirect, BrowserRouter } from 'react-router-dom';
import Login from './Login';
import Home from './Home';
import '../style/App.scss';

const App = () => {
	const [isLoggedIn, setIsLoggedIn] = useState(false);
	const [id, setId] = useState('');
	return (
		<BrowserRouter>
			<div className="app">
				<Switch>
					<Route path='/home'>
						<Home id={id} />
					</Route>
					<Route 
						path='/' 
						render={props => (isLoggedIn ?
							<Redirect to='/home' /> : 
							<Login {...props} isLoggedIn={isLoggedIn} setIsLoggedIn={setIsLoggedIn} setId={setId} />
						)}
					/>
				</Switch>
			</div>
		</BrowserRouter>
	);
}

export default App;
