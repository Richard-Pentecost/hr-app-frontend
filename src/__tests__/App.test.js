// import { render, screen } from '@testing-library/react';
import React from 'react';
import Enzyme from 'enzyme';
import App from '../components/App';

describe('App', () => {
  it('renders a div with the className = "app"', () => {
    const wrapper = Enzyme.shallow((
      <App />
    ));
    expect(wrapper.find('div').hasClass('app')).toBe(true)
  });
})
