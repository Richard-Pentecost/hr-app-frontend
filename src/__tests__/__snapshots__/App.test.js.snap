// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<BrowserRouter>
  <div
    className="app"
  >
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <AuthRoute
        component={[Function]}
        exact={true}
        id=""
        isLoggedIn={false}
        path="/home"
      />
    </Switch>
  </div>
</BrowserRouter>
`;
